roadmap/implementers-guide/src/node/backing/statement-distribution.md

```diff
@@ -1,107 +1,453 @@
+    -   Why: spam prevention
+    -   Validators can try to circumvent this, but they would only consume a few KB
+        of memory and it is trivially slashable on chain.
+-   `ClusterTracker`
+    -   Determine whether to accept/reject messages from other validators in the
+        same group
+    -   Keep track of what we have sent to other validators in the group and pending
+        statements.
+-   Protocol: See "Protocol"
+
+## Grid
+
+-   Distribution of backed candidates and late statements outside the group.
+-   Protocol: See "Protocol"
+
+### Grid topology
+
+-   The gossip topology
+    -   Why: limit the amount of peers we send messages to and handle view updates.
+-   The basic operation of the 2D grid topology is that:
+    -   A validator producing a message sends it to its row-neighbors and its
+        column-neighbors
+    -   A validator receiving a message originating from one of its row-neighbors
+        sends it to its column-neighbors
+    -   A validator receiving a message originating from one of its column-neighbors
+        sends it to its row-neighbors
+-   This grid approach defines 2 unique paths for every validator to reach every
+    other validator in at most 2 hops.
+-   However, we also supplement this with some degree of random propagation.
+    -   Why: insert some redundancy, protect against targeted attacks.
+
+### Grid utilities
+
+-   `SessionGridTopology`
+    -   Topology representation for a session.
+-   `SessionTopologyView`
+    -   Our local view of the grid topology for a session.
+    -   Built from `SessionGridTopology`, a list of groups and our validator index.
+    -   Propagation
+        -   For groups that we are in, receive from nobody and send to our X/Y peers.
+        -   For groups that we are not part of:
+            -   We receive from any validator in the group we share a slice with and
+                send to the corresponding X/Y slice in the other dimension.
+            -   For any validators we don't share a slice with, we receive from the
+                nodes which share a slice with them.
+-   `GroupSubView`
+    -   Our local view of a subset of the grid topology organized around a specific
+        validator group.
+-   `GridTracker`
+    -   A tracker of knowledge from authorities within the grid for a particular
+        relay parent.
+-   `GridTracker::import_manifest`
+    -   Called when: receiving an incoming manifest (full or partial).
+    -   Attempts to import a manifest advertised by a remote peer.
+    -   Checks whether the peer is allowed to send us manifests about this group at
+        this relay parent.
+    -   Does sanity checks on the manifest itself.
+    -   On success, returns whether an acknowledgement should be sent in response.
+        -   Only occurs when the candidate is already known to be confirmed backed,
+            and the validator is in the set we are receiving from.
```

Comment:
These function docs are high-level summaries of what the docstrings already say. But it helped me to put them in one place so I could reference them without jumping around the code. I'm thinking of keeping a documentation of the main API entrypoints of each module, but just shortening them to a one-line description each. Thoughts appreciated.

Reply:
I think it's better to describe behavior rather than code structure - to that extent, this is good. But I would prefer to omit the details of `GridTracker` being used to implement this. It's too proscriptive, and too likely to get out of date.